### PR TITLE
Feature/patro23 24 list of users in application

### PR DESCRIPTION
**Pull request by a-lipert**
https://tracker.intive.com/jira/browse/PATRO23-24

Cooperation with @pzajaczkowski.

### CODE REVIEW

**Comment by pzajaczkowski** [comment]
Keycloak API search works like sql -> LIKE 'a%'. So our endpoint search works the same.

File: src/modules/user/application/GettingUsers/GetUsers.cs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+	public string? Search { get; set; }
+
+	/// <inheritdoc/>
+	public List<SortDescriptor> SortDescriptors { get; set; } = null!;
+}
+
+/// <summary>
+/// Get Users handler.
+/// </summary>
+public class GetUsersQueryHandler : IQueryHandler<GetUsers, PagedList<UserInfo>>
```
Comment (mateuszgiedzinski) [inline src/modules/user/application/GettingUsers/GetUsers.cs:32]: So this Query Handler is independent from database so we can write unit tests for it.

**Review by mateuszgiedzinski — CHANGES_REQUESTED** [review]

File: src/modules/user/contracts/UserAttributes.cs [hunk 2]
```diff
@@ -8,5 +8,5 @@ public class UserAttributes
 	/// <summary>
 	/// User avatar identifier.
 	/// </summary>
-	public string Avatar { get; set; } = null!;
+	public string[] Avatar { get; set; } = null!;
```
Comment (IzabelaPalubicka) [inline src/modules/user/contracts/UserAttributes.cs:11]: Why is this an array?

File: src/modules/user/api/UserModule.cs [hunk 3]
```diff
@@ -12,11 +14,13 @@ public static class UserModule
...
-	/// <param name="configurationManager">ConfigurationManager.</param>
 	/// <returns>Updated IServiceCollection.</returns>
-	public static IServiceCollection AddUserModule(this IServiceCollection services, ConfigurationManager configurationManager)
+	public static IServiceCollection AddUserModule(this IServiceCollection services)
 	{
-		services.AddScoped<IExecutionContextAccessor, ExecutionContextAccessor>();
+		services.AddScoped<IValidator<SignInUser>, SignInUserValidator>();
+		services.AddScoped<IValidator<CreateUser>, CreateUserValidator>();
+		services.AddScoped<IValidator<GetUsers>, GetUsersValidator>();
+		//// services.AddScoped<IExecutionContextAccessor, ExecutionContextAccessor>();
```
Comment (IzabelaPalubicka) [inline src/modules/user/api/UserModule.cs:23]: comment to delete?

File: src/modules/user/application/GettingUsers/GetUsersValidator.cs [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+/// </summary>
+public class GetUsersValidator : AbstractValidator<GetUsers>
+{
+	/// <summary>
+	/// Initializes a new instance of the <see cref="GetUsersValidator"/> class.
+	/// </summary>
+	public GetUsersValidator()
+	{
+		this.RuleFor(budget => budget.PageIndex).GreaterThan(0);
+		this.RuleFor(budget => budget.PageSize).GreaterThan(0);
```
Comment (IzabelaPalubicka) [inline src/modules/user/application/GettingUsers/GetUsersValidator.cs:16]: change budget to x

Comment (mateuszgiedzinski) [reply src/modules/user/application/GettingUsers/GetUsersValidator.cs:16]: someone tells me that we should use x, y, n  in this order as lambda names but I sometimes use item also. But in this case of course budget is invalid.

Comment (a-lipert) [reply src/modules/user/application/GettingUsers/GetUsersValidator.cs:16]: I don't know how we overlooked that :D thanks

File: src/modules/user/contracts/UserAttributes.cs (hunk 2 above)
Comment (pzajaczkowski) [reply src/modules/user/contracts/UserAttributes.cs:11]: Due to json deserialization. That's what we are receiving from Keycloak Rest API.

File: src/modules/user/application/GettingUsers/GetUsersValidator.cs (hunk 4 above)
Comment (pzajaczkowski) [reply src/modules/user/application/GettingUsers/GetUsersValidator.cs]: well known copy paste technique xD

File: src/modules/user/api/UserModule.cs (hunk 3 above)
Comment (a-lipert) [reply src/modules/user/api/UserModule.cs]: It was marked to be moved to shared module. It's done now.

**Review by IzabelaPalubicka — APPROVED** [review]
Works fine :smiley:

File: src/modules/user/application/GettingUsers/GetUsers.cs (hunk 1 above)
Comment (a-lipert) [reply src/modules/user/application/GettingUsers/GetUsers.cs:32]: Done :)

File: tests/unit/Intive.Patronage2023.User.Application.Tests/GetUsersQueryHandlerTests.cs [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+		// Assert
+		responseSortedUsers.Should().Equal(fakedSortedUsers);
+	}
+
+	/// <summary>
+	/// Test that checks if the pagination is applied for returned values.
+	/// </summary>
+
+	[Fact]
+	public async Task Handle_WhenCalled_ShouldApplyPaginationForUsersList()
```
Comment (mateuszgiedzinski) [inline tests/unit/Intive.Patronage2023.User.Application.Tests/GetUsersQueryHandlerTests.cs:82]: When + conditions under what you tests for example WhenPaginationSpecified

**Review by mateuszgiedzinski — CHANGES_REQUESTED** [review]

File: tests/unit/Intive.Patronage2023.User.Application.Tests/GetUsersQueryHandlerTests.cs [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+		var query = new GetUsers() { PageSize = pageSize, SortDescriptors = sortDescriptors };
+
+		// Act and Assert
+		await Assert.ThrowsAsync<NotSupportedException>(async () => await this.handler.Handle(query, CancellationToken.None));
+	}
+
+	/// <summary>
+	/// Test that checks if the pagination, if specified, is applied for returned values.
+	/// </summary>
+
```
Comment (mateuszgiedzinski) [inline tests/unit/Intive.Patronage2023.User.Application.Tests/GetUsersQueryHandlerTests.cs:97]: empty line is not needed. I suggest using CTRL+K, CTRL+E shortcut to cleanup active doc.

**Review by mateuszgiedzinski — APPROVED** [review]
Only one line break needs fix. But it's minor change.

**Pull request merged**